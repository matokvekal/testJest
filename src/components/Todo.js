import React from "react";
////1////////////
// function Todo() {

//   return <div data-testid="test-3">Hello world!!</div>;
// }

/////2///////////
function Todo( {todo} ) {

  const { id, title, completed } = todo;
  const H1 = <h1>todo -----{title}</h1>;
  const text = completed ? <strike>{H1}</strike> : H1;
  return (
    <>
      {/* <div data-testid={`todo-${id}`}>{text}</div> */}
      <div data-testid={`todo-${id}`}>--{text}--</div> {/* press u to update snapshot*/}
    </>
  );
}

  export default Todo;
