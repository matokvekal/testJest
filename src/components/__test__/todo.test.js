import { render, screen, cleanup } from "@testing-library/react";
import Todo from "../../components/Todo";
import renderer from "react-test-renderer";

////1/////
// test("test1", () => {
//   expect(1 + 1).toBe(2);
// });
// test("test1", () => {
//   expect(3 + 1).toBe(4);
// });

// test("test2", () => {
//   expect(true).toBe(true);
// });


/////2//////
// test("should render todo component", () => {
//    render(<Todo />);
//    const todoElement = screen.getByTestId("test-3");
//    expect(todoElement).toBeInTheDocument();
//    expect(todoElement).toHaveTextContent("Hello world!");
//  });

////3/////

afterEach(()=>{
   cleanup();
})

// test("should render non-component todo", () => {
//    const todo = {id:1, title:"bring milk", completed:false,};
//    render(<Todo todo={todo}/>);
//    const todoElement = screen.getByTestId("todo-1");
//    expect(todoElement).toBeInTheDocument();
//    expect(todoElement).toHaveTextContent("bring milk");
//    expect(todoElement).not.toContainHTML("<strike>");
//  });

//  test("should render completed todo", () => {
//    const todo = {id:2, title:"wash dishes", completed:true,};
//    render(<Todo todo={todo}/>);
//    const todoElement = screen.getByTestId("todo-2");
//    expect(todoElement).toBeInTheDocument();
//    expect(todoElement).toHaveTextContent("wash dishes");
//    expect(todoElement).toContainHTML("strike");
//  });

/////////4//////////

test("matches snapshot", () => {
  const todo = { id: 1, title: "bring milk", completed: false };
  const tree = renderer.create(<Todo todo={todo} />).toJSON();
  console.log("tree", tree);
  expect(tree).toMatchSnapshot();
});
