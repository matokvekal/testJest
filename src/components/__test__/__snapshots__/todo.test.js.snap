// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
Array [
  <div
    data-testid="todo-1"
  >
    --
    <h1>
      todo -----
      bring milk
    </h1>
    --
  </div>,
  " ",
]
`;
